src/sunadaptcontroller/soderlind/sunadaptcontroller_soderlind.c

```diff
@@ -117,6 +118,16 @@ SUNErrCode SUNAdaptController_SetParams_Soderlind(SUNAdaptController C,
   SODERLIND_K3(C) = k3;
   SODERLIND_K4(C) = k4;
   SODERLIND_K5(C) = k5;
+
+  if (k5 != SUN_RCONST(0.0) || k3 != SUN_RCONST(0.0))
+  {
+    SODERLIND_HISTORYSIZE(C) = 2;
+  }
+  else if (k4 != SUN_RCONST(0.0) || k5 != SUN_RCONST(0.0))
```

Comment:
Should this check `k2` also?
```suggestion
  else if (k4 != SUN_RCONST(0.0) || k5 != SUN_RCONST(0.0) || k2 != SUN_RCONST(0.0))
```

Reply:
Good catch. It should check `k2` instead of `k5`

Reply:
Should this still check `k5` i.e., `k2` and `k4` are are zero (not that such a setup really makes sense)?

Reply:
There's no need since it would have entered the previous if statement if k5 was nonzero